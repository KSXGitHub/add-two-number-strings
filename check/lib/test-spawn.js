'use strict'

function main ({
  path = require('path'),
  childProcess: {spawnSync} = require('child_process'),
  process: {env} = require('process'),
  alwaysPrintStdIO = false,
  defaultExecutable = 'echo',
  envMiddleName = '?'
} = {}) {
  const {
    [`JEST_${envMiddleName}_EXECUTABLE`]: executable = defaultExecutable,
    [`JEST_${envMiddleName}_ARGV`]: spawnArguments = '[]',
    [`JEST_${envMiddleName}_SKIP`]: skipSpawnTesting = 'false'
  } = env

  const wdir = path.resolve(__dirname, '../..')

  test('JavaScript Code Style: StandardJS', () => {
    if (skipSpawnTesting.toLowerCase() === 'true') return

    const argv = JSON.parse(spawnArguments)
    expect(argv).toBeInstanceOf(Array)

    const {
      stdout,
      stderr,
      signal,
      error,
      status
    } = spawnSync(executable, argv, {cwd: wdir, shell: true})

    if (stdout === null) console.warn('respose.stdout is null')
    if (stderr === null) console.warn('respose.stderr is null')
    if (signal) console.warn(`respose.signal is ${JSON.stringify(signal)}`)
    if (error) throw error
    if (status) throw new Error(stderr + '\n' + stdout)
    if (alwaysPrintStdIO) console.log(stderr + '\n' + stdout)
  })
}

module.exports = main
