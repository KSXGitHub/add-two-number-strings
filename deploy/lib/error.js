'use strict'

const {freeze} = Object

class ServerError extends Error {
  constructor (message, response) {
    super(message)
    return freeze({
      message,
      response,
      __proto__: this
    })
  }
  get name () {
    return 'ServerError'
  }
}

class GitHubError extends ServerError {
  get name () {
    return 'GitHubError'
  }
}

module.exports = {ServerError, GitHubError}
