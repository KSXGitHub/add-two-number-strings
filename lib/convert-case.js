'use strict'

const REGEX = {
  WORD_DELIM: /[_\- \t\n\r]+/,
  NEWLINE: /\r?\n\r?/
}

const convertCase = {
  toCamelCase: (string = '') => {
    const [first, ...rest] = String(string).split(REGEX.WORD_DELIM)

    return [
      first.toLowerCase(),
      ...rest.map(convertCase.capitalize)
    ].join('')
  },

  toPascalCase: (string = '') =>
    String(string).split(REGEX.WORD_DELIM).map(convertCase.capitalize).join(''),

  toKebabCase: (string = '') =>
    convertCase.seperaterize(string, convertCase.lowerize, '-'),

  toUpperKebabCase: (string = '') =>
    convertCase.seperaterize(string, convertCase.upperize, '-'),

  toSnakeCase: (string = '') =>
    convertCase.seperaterize(string, convertCase.lowerize, '_'),

  toUpperSnakeCase: (string = '') =>
    convertCase.seperaterize(string, convertCase.upperize, '_'),

  seperaterize: (string = '', convertCase, delim) =>
    String(string).split(REGEX.WORD_DELIM).map(convertCase).join(delim),

  lowerize: (string = '') => String(string).toLowerCase(),

  upperize: (string = '') => String(string).toUpperCase(),

  capitalize: ([first, ...rest] = '') => [
    String(first).toUpperCase(),
    ...rest.map(x => String(x).toLowerCase())
  ].join('')
}

const proto = {
  REGEX,
  convertCase
}

module.exports = Object.assign(Object.create(proto), REGEX, convertCase)
