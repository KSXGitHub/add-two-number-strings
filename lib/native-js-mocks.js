'use strict'

global.Date = class MockedDate extends Date {
  constructor (data = 0, ...args) {
    super(data, ...args)
  }

  toString () {
    return this.toLocaleTimeString()
  }

  toLocaleString () {
    return this.toString()
  }

  toTimeString () {
    return this.toLocaleTimeString()
  }

  toDateString () {
    return this.toLocaleDateString()
  }

  toLocaleTimeString () {
    return super.toLocaleTimeString('en-US', {timeZone: 'UTC'})
  }

  toLocaleDateString () {
    return super.toLocaleDateString('en-US', {timeZone: 'UTC'})
  }
}

global.Math = Object.assign(
  Object.create(Math),
  Math,
  {random: () => 0.8253283060883272}
)
