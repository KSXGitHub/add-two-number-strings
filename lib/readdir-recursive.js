'use strict'
const {join} = require('path')
const {Yield} = require('x-iterable').createClass

const yieldof = fn =>
  (...args) => new Yield(fn(...args))

const readdirRecursiveSync = yieldof(
  function * fn (dir = '', fs = require('fs')) {
    const {readdirSync, statSync} = fs
    for (const item of readdirSync(dir)) {
      const fullpath = join(dir, item)
      yield fullpath
      if (statSync(fullpath).isDirectory()) yield * fn(fullpath)
    }
  }
)

module.exports = {
  readdirRecursiveSync
}
