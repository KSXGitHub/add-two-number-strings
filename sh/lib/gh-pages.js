'use strict'
const {resolve} = require('path')
const {exit} = require('process')
const ghpages = require('gh-pages')

ghpages.publish(
  resolve(__dirname, '../../dist'),
  {
    dotfiles: true,
    message: 'Update using gh-pages'
  },
  error => {
    if (!error) return
    console.error(error)
    exit(1)
  }
)
