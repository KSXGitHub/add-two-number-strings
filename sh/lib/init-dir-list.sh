echo ./out
echo ./release
echo ./object
