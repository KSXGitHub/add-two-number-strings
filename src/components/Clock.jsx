import React from 'react'

export default class Clock extends React.Component {
  constructor (props) {
    super(props)

    const {
      refreshRate = 1000,
      format: {
        wrap: Wrap = props => (<span {...props} />),
        content: getContent = date => date.toTimeString()
      } = {},
      display = date => (<span>{getContent(date)}</span>)
    } = props

    this.state = {date: new Date(), display}
    this.refreshRate = parseInt(refreshRate)
  }

  render () {
    return this.state.display(this.state.date)
  }

  componentDidMount () {
    this.timerID = setInterval(() => this.tick(), this.refreshRate)
  }

  componentWillUnmount () {
    clearInterval(this.timerID)
    delete this.timerID
  }

  tick () {
    this.setState({date: new Date()})
  }
}
