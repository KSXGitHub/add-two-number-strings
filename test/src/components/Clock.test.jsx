import './__lib__/material-ui-mocks.jsx'
import '../../../lib/native-js-mocks'
import React from 'react'
import renderer from 'react-test-renderer'
import Clock from '../../../src/components/Clock.jsx'

describe('Snapshot: components/Clock.jsx', () => {
  const display = date => (<p>{String(date)}</p>)
  const wrap = props => (<p {...props} />)
  const content = date => String(date)

  const suite = {
    'props = {}': (<Clock />),
    'props = {display}': (<Clock
      display={display}
    />),
    'props.format = {wrap}': (<Clock
      format={{wrap}}
    />),
    'props.format = {content}': (<Clock
      format={{content}}
    />),
    'props.format = {wrap, content}': (<Clock
      format={{wrap, content}}
    />)
  }

  for (const title in suite) {
    test(title, () => {
      const tree = renderer.create(suite[title]).toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
