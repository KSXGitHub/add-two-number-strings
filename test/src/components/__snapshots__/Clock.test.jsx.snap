// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: components/Clock.jsx props = {} 1`] = `
<span>
  07:00:00 GMT+0700 (+08)
</span>
`;

exports[`Snapshot: components/Clock.jsx props = {display} 1`] = `
<p>
  12:00:00 AM
</p>
`;

exports[`Snapshot: components/Clock.jsx props.format = {content} 1`] = `
<span>
  12:00:00 AM
</span>
`;

exports[`Snapshot: components/Clock.jsx props.format = {wrap, content} 1`] = `
<span>
  12:00:00 AM
</span>
`;

exports[`Snapshot: components/Clock.jsx props.format = {wrap} 1`] = `
<span>
  07:00:00 GMT+0700 (+08)
</span>
`;
